### ISSUE TITLE
Add rule to whitelist allowed languages in code fences

### ISSUE THREAD

**Issue by Lunkentuss** [DavidAnson/markdownlint#670]
I often find that languages in code fences contains typos or various alternative spellings, such as "yml" and "yaml" across the repository. This can have critical consequences when tools will only enable syntax highlighting for certain spellings. Therefor it would be nice to have a rule, which is disabled by default to allow backwards compatability, to only allow certain languages.

```yaml
---
# Default configuration (disabled by default)
MD0XX: false

---
# Override default
MD0XX:
  whitelist: ["yaml", "java", "javascript"]
```

With the overridden rule, using a fence block such as

~~~md
```my-non-whitelisted-language
lorem ipsum
```
~~~

will fail. I don't know if it makes sense to incorporate this with rule MD040 or not.

**Comment by Lunkentuss** [comment]
I'm closing this since I realized a similar issue already exists: https://github.com/DavidAnson/markdownlint/issues/670
